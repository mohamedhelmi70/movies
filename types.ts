export type RootParamList = {
    Movies: undefined;
    Movie: {
        movieId: number;
        movie: Movie;
    };
};

export interface Movie {
    poster_path: string;
    adult?: boolean;
    overview: string;
    release_date: string,
    genre_ids: number[],
    id: number,
    original_title: string,
    original_language?: string,
    title: string,
    backdrop_path?: string,
    popularity?: number,
    vote_count?: number,
    video?: boolean,
    vote_average: number,
    genres?: Genre[]
}

export interface Genre {
    id: number;
    name: string;
}

export interface Credit {
    adult?: boolean,
    gender?: number,
    id: number,
    known_for_department?: string,
    name: string,
    original_name: string,
    popularity?: number,
    profile_path: string,
    cast_id?: number,
    character?: string,
    credit_id: string,
    order?: number
}