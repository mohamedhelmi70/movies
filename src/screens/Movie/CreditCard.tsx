import React from 'react';
import {Image, StyleSheet} from 'react-native';
import {Credit} from '../../../types';
import {userDefault} from '../../assets';
import {FastImage} from '../../components';
import {Text, View} from '../../components/Themed';
import {wp} from '../../constants/Layout';

interface CreditCardProps {
    credit: Credit;
}

function CrditCard({credit}: CreditCardProps) {
    return (
        <View style={styles.card}>
            <View>
                {Boolean(credit?.profile_path) ? (
                    <FastImage
                        url={`https://www.themoviedb.org/t/p/original${credit?.profile_path}`}
                        style={styles.profile}
                    />
                ) : (
                    <Image source={userDefault} style={styles.profile} resizeMode="cover" />
                )}
            </View>
            <Text weight="600" font="Roboto-Medium" style={styles.cardTitle}>
                {credit?.name}
            </Text>
        </View>
    );
}

const styles = StyleSheet.create({
    card: {
        flex: 1,
        marginRight: wp('7%'),
        borderRadius: wp('2%'),
        backgroundColor: '#fff',
        justifyContent: 'flex-start',
        alignItems: 'flex-start',
    },
    profile: {
        width: wp('20%'),
        height: wp('20%'),
        borderRadius: wp('10%'),
    },
    cardTitle: {
        marginTop: wp('1%'),
        fontSize: wp('3%'),
        maxWidth: wp('27%'),
    },
});

export default CrditCard;
