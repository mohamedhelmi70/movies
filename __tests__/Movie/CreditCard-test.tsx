import React from 'react';
import renderer from 'react-test-renderer';
import {Credit} from '../../types';
import CreditCard from '../../src/screens/Movie/CreditCard';

test('renders CreditCard correctly', () => {
    const creditItem: Credit = {
        id: 123,
        name: 'test',
        original_name: 'test',
        profile_path: '/profile.png',
        credit_id: 'cridet_123',
    };
    const tree = renderer.create(<CreditCard credit={creditItem} />).toJSON();
    expect(tree).toMatchSnapshot();
});
