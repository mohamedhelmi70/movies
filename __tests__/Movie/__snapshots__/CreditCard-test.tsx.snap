// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CreditCard correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FFF",
      },
      Object {
        "alignItems": "flex-start",
        "backgroundColor": "#fff",
        "borderRadius": 15,
        "flex": 1,
        "justifyContent": "flex-start",
        "marginRight": 52.5,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FFF",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 75,
            "height": 150,
            "width": 150,
          },
        ]
      }
    >
      <FastImageView
        resizeMode="cover"
        source={
          Object {
            "cache": "immutable",
            "headers": Object {
              "Authorization": "4f298a53e552283bee957836a529baec",
            },
            "priority": "normal",
            "uri": "https://www.themoviedb.org/t/p/original/profile.png",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
          "fontFamily": "Roboto",
          "fontWeight": "600",
          "textAlign": "left",
        },
        Object {
          "fontSize": 22.5,
          "marginTop": 7.5,
          "maxWidth": 202.5,
        },
      ]
    }
  >
    test
  </Text>
</View>
`;
